src/dev/build/tasks/build_packages_task.ts

```diff
@@ -255,7 +255,7 @@ export const BuildPackages: Task = {
               Path.resolve(pkgDistPath, 'package-map.json'),
               JSON.stringify(
                 packages
-                  .filter((p) => p.isPlugin())
+                  .filter((p) => p.isPlugin() && !p.isDevOnly())
```

Comment:
__note:__ without this change, `@kbn/mock-idp-plugin` (dev-only) is included into `package-map.json` even though it's not actually bundled into the build. Let me know if you have any better ideas how to solve this!

Reply:
Is this not going to interfere with other builds? 
Is it related to this thread: https://github.com/elastic/kibana/pull/172257#discussion_r1424957229? 

@mistic , do you have any opinions on this?

Reply:
@azasypkin I suggest we change that line https://github.com/elastic/kibana/blob/main/src/dev/build/lib/config.ts#L249 with `(!p.isPlugin() || (this.pluginFilter(p) && !p.isDevOnly()))` instead of adding the check on this line

Reply:
Sure, @mistic. Applied your suggestion in https://github.com/elastic/kibana/pull/172257/commits/f979aa7a006f698822271af4fb97bd672989b112.